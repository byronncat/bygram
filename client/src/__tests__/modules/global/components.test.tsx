import rerender from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'
import { Loader, Overlay } from '@global'

describe('components', () => {
  describe('Loader', () => {
    it('should render the loader correctly', () => {
      const loader = rerender.create(<Loader />)
      expect(loader).toMatchSnapshot()
    })
  })

  describe('Overlay', () => {
    it('should render the overlay correctly', async () => {
      const exitHandler = jest.fn()

      const overlay = rerender.create(
        <Overlay exitHandler={exitHandler}>
          <div>Test</div>
        </Overlay>
      )
      expect(overlay).toMatchSnapshot()
    })

    it('should call the exitHandler when the overlay is clicked', async () => {
      const exitHandler = jest.fn()

      const { container } = await render(
        <Overlay exitHandler={exitHandler}>
          <div>Test</div>
        </Overlay>
      )

      const overlay = container.querySelector('.overlay')
      if (overlay) await fireEvent.click(overlay)
      const exitButton = container.querySelector('button')
      if (exitButton) await fireEvent.click(exitButton)

      expect(exitHandler).toHaveBeenCalledTimes(2)
    })
  })
})
