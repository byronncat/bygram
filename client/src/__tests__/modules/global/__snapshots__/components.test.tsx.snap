// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Loader should render the loader correctly 1`] = `
<div
  className="d-flex flex-column justify-content-center align-items-center w-100 h-100"
>
  <div
    className="spinner-box"
  >
    <div
      className="configure-border-1"
    >
      <span
        className="configure-core"
      />
    </div>
    <div
      className="configure-border-2"
    >
      <span
        className="configure-core"
      />
    </div>
  </div>
  <div
    className="text waviy mt-5"
  >
    <span>
      L
    </span>
    <span>
      O
    </span>
    <span>
      A
    </span>
    <span>
      D
    </span>
    <span>
      I
    </span>
    <span>
      N
    </span>
    <span>
      G
    </span>
  </div>
</div>
`;

exports[`components Overlay should render the overlay correctly 1`] = `
<div
  className="position-absolute top-0 start-0 z-1 w-100 h-100 d-flex justify-content-center align-items-center"
>
  <span
    className="overlay z-n1"
    onClick={[MockFunction]}
  />
  <button
    aria-label="close"
    className="button shadow-none btn-close btn-close-white position-absolute top-0 end-0 p-3"
    onClick={[MockFunction]}
  />
  <div>
    Test
  </div>
</div>
`;
